clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -8857,7 +8856,7 @@ def err_nontemporal_builtin_must_be_pointer_intfltptr_or_vector : Error<
   "address argument to nontemporal builtin must be a pointer to integer, float, "
   "pointer, or a vector of such types (%0 invalid)">;
 
-def err_deleted_function_use : Error<"attempt to use a deleted function">;
+def err_deleted_function_use : Error<"attempt to use a deleted function%select{|: %1}0">;
```

Comment:
I think we should make the user-provided message part of a note (and keep the error unchanged)

Reply:
Yeah, I’ve considered that as well; my reasoning here was that the error message is likely what users of this feature would primarily want us to display, and adding it to the error proper rather than just to the note makes it more obvious that this is the message you should pay attention to, if you will (consider also that, iirc, we’re also displaying `static_assert` messages in the error instead of in a note).

Also, as a side note, we already have three or so different notes for this, so we’d either have to edit all of those or add another one; the former is just as much work as changing the errors, and the latter would just add more clutter, in my opinion at least.

Reply:
@AaronBallman @erichkeane for additional opinions

Reply:
I think it should mirror static assert messages, which is the colon + message in the error.  It is 'important' enough IMO to be considered a part of the first diagnostic.

Reply:
> Also, as a side note, we already have three or so different notes for this, so we’d either have to edit all of those or add another one; the former is just as much work as changing the errors, and the latter would just add more clutter, in my opinion at least.

Had to change quite a few more diagnostics than I had anticipated, but I’d still argue that while changing the notes would probably be easier, it still makes more sense to display the message in the error proper because that’s what users are most likely to pay attention to (and in some cases, e.g. if the error is displayed in a popup in an IDE, users might not actually be shown any of the notes).